modules/nf-commons/src/main/nextflow/util/CmdLineOptionMap.groovy

```diff
@@ -0,0 +1,63 @@
+package nextflow.util
+
+/**
+ * Holder for parsed command line options.
+ *
+ * @author Manuele Simi <manuele.simi@gmail.com>
+ */
+class CmdLineOptionMap {
+
+    final options = new LinkedHashMap<String, List<String>>()
+    final static instance = new CmdLineOptionMap()
+
+    protected addOption(String key, String value) {
+        if ( !options.containsKey(key) )
+            options[key] = new ArrayList<String>()
+        options[key].add(value)
+    }
+
+    boolean hasMultipleValues(String key) {
+        options.containsKey(key) ? options[key].size() > 1 : false
+    }
+
+    boolean hasOptions() {
+        options.size()
+    }
+
+    def getValues(String key) {
```

Comment:
Does this always return a `List`? in this case worth declaring `List getValues(String key)`

Reply:
done